libp2p/protocols/secure/secure.nim

```diff
@@ -120,10 +120,23 @@ method readOnce*(s: SecureConn,
   doAssert(nbytes > 0, "nbytes must be positive integer")
 
   if s.buf.data().len() == 0:
-    let buf = await s.readMessage()
+    let (buf, err) = try:
+        (await s.readMessage(), nil)
+      except CatchableError as exc:
+        (@[], exc)
```

Comment:
I think it would be better to return `(@[], exc.name, exc.msg)` because i'm not sure about lifetime of this exception object.

Reply:
Hmm I kinda wanted to just re-throw, what makes you think the reference would be collected?

Reply:
will just copy the ref (add_ref) and exit the block

Reply:
I tested it a bit, seems to be fine